src/functions/types.ts

```diff
@@ -396,3 +397,40 @@ export type RuntimeUnhandledEventContext<
     // deno-lint-ignore no-explicit-any
     body: any;
   };
+
+/**
+ * @description Used to extract returned args from Async and Sync function handlers
+ */
+type FunctionRuntimeReturnArgs<
```

Comment:
This type is simply used to isolate the logic required to extract `FunctionRuntimeReturnArgs` which is the returned type from `AsyncFunctionHandler` or `SyncFunctionHandler`
`AsyncFunctionHandler` => `Promise<FunctionHandlerReturnArgs<OutputParameters>>`
`SyncFunctionHandler` => `FunctionHandlerReturnArgs<OutputParameters>`

I've rename the type to `ExtractFunctionRuntimeReturnArgs` because it handles the exaction for Async and Sync function types

Reply:
I like that